gluten-data/src/main/java/io/glutenproject/columnarbatch/ColumnarBatches.java

```diff
@@ -323,7 +323,7 @@ public static ColumnarBatch create(ExecutionCtx ctx, long nativeHandle) {
     int numColumns = Math.toIntExact(iv.getNumColumns());
     int numRows = Math.toIntExact(iv.getNumRows());
     if (numColumns == 0) {
-      return new ColumnarBatch(new ColumnVector[0], numRows);
+      return new ColumnarBatch(new ColumnVector[] {iv}, numRows);
```

Comment:
If the ColumnarBatch is light, then it must have one column. I think it won't break anything, but better to double check cc @zhztheplayer @PHILO-HE

Reply:
What happened without this change?

Reply:
`count(1)` requires child empty output, so the selected column for table cache is 0. Without this change, we can not make a valid light ColumnarBatch.

Reply:
A batch with `new ColumnVector[] {iv}` creates inconsistency since it's both 1 column (by ColumnarBatch#numCols()) and 0 column (by iv.getNumColumns()) at the same time. So I am trying to understand what happened here. 

If it's about the call to `ColumnarBatches.create` in file `ColumnarCachedBatchSerializer.scala`, can we just improve `ColumnarBatches.select` to make the method support zero-column batch?

Also I think it's safe to consider a zero-column batch a heavy batch. Although our code didn't do that yet.

Besides FWIW the whole "light" "heavy" thing can be removed after stopping support Spark 3.2, there was a issue to track that https://github.com/oap-project/gluten/issues/540.

Reply:
>  If it's about the call to ColumnarBatches.create in file ColumnarCachedBatchSerializer.scala, can we just improve ColumnarBatches.select to make the method support zero-column batch?

It seems we can not.. Once we want to create a new light ColumnarBatch, we need to inject the `IndicatorVector`. That means we must have one column even if the acutally selected column size is 0.

> Also I think it's safe to consider a zero-column batch a heavy batch. Although our code didn't do that yet.

Yes, we can, and this method is only used to create light ColumnarBatch which should always have a `IndicatorVector`. So I think this change won't break this supposition.

> Besides FWIW the whole "light" "heavy" thing can be removed after stopping support Spark 3.2, there was a issue to track that https://github.com/oap-project/gluten/issues/540.

I see it, use a custom ColumnarBatch is better in future.

Reply:
Thanks for the explanation. Let's use your code temporarily until finding a better way.